packages/hardhat-core/test/builtin-tasks/compile.ts

```diff
@@ -46,6 +48,41 @@ describe("compile task", function () {
     );
   }
 
+  describe("compile with latest solc version", function () {
+    // The 'hardhat.config.js' and 'A.sol' files need to be updated each time a new solc version is released
+
+    useFixtureProject("compilation-latest-solc-version");
+    useEnvironment();
+
+    it("should have the last version of solc in the 'hardhat.config.js' and 'A.sol' files", async function () {
+      // Test to check that the last version of solc is being tested
+      const userConfigSolcVersion =
+        HardhatContext.getHardhatContext().environment?.userConfig.solidity;
```

Comment:
You should be able to replace `HardhatContext.getHardhatContext().environment` with just `this.env`